frappe/core/doctype/communication/communication.js

```diff
@@ -209,10 +219,18 @@ frappe.ui.form.on("Communication", {
 		var first_name = names[0]
 		var last_name = names.length >= 2? names[names.length - 1]: ""
 
-		frappe.route_options = {
-			"email_id": frm.doc.sender,
-			"first_name": first_name,
-			"last_name": last_name,
+		if(frm.doc.communication_medium=="Email"){
```

Comment:
if - else is redundant, pass all the options (nulls will be ignored)
```
frappe.route_options = {
	"email_id": frm.doc.sender,
	"first_name": first_name,
	"last_name": last_name,
	"mobile_no": frm.doc.phone_no
}
```
I hope you are not being paid by LOC 😉

Reply:
Sorry for that and thank you